src/js/Draw/L.PM.Draw.js

```diff
@@ -219,6 +219,42 @@ const Draw = L.Class.extend({
         (this._map.pm.globalOptions.panes &&
           this._map.pm.globalOptions.panes.markerPane) ||
         'markerPane';
+    } else if (type === 'polygonPane') {
+      layer.options.pane =
+        (this._map.pm.globalOptions.panes &&
+          (this._map.pm.globalOptions.panes.polygonPane ||
+            this._map.pm.globalOptions.panes.layerPane)) ||
+        'overlayPane';
+    } else if (type === 'circlePane') {
+      layer.options.pane =
+        (this._map.pm.globalOptions.panes &&
+          (this._map.pm.globalOptions.panes.circlePane ||
+            this._map.pm.globalOptions.panes.layerPane)) ||
+        'overlayPane';
+    } else if (type === 'circleMarkerPane') {
+      layer.options.pane =
+        (this._map.pm.globalOptions.panes &&
+          (this._map.pm.globalOptions.panes.circleMarkerPane ||
+            this._map.pm.globalOptions.panes.layerPane)) ||
+        'overlayPane';
+    } else if (type === 'polylinePane') {
+      layer.options.pane =
+        (this._map.pm.globalOptions.panes &&
+          (this._map.pm.globalOptions.panes.polylinePane ||
+            this._map.pm.globalOptions.panes.layerPane)) ||
+        'overlayPane';
+    } else if (type === 'rectanglePane') {
+      layer.options.pane =
+        (this._map.pm.globalOptions.panes &&
+          (this._map.pm.globalOptions.panes.rectanglePane ||
+            this._map.pm.globalOptions.panes.layerPane)) ||
+        'overlayPane';
+    } else if (type === 'textPane') {
+      layer.options.pane =
+        (this._map.pm.globalOptions.panes &&
+          (this._map.pm.globalOptions.panes.textPane ||
+            this._map.pm.globalOptions.panes.markerPane)) ||
+        'markerPane';
```

Comment:
Let us simplify this a little bit:
```suggestion
_setPane(layer, type) {
    const panes = this._map.pm.globalOptions.panes || {};
    if (type === 'layerPane') {
      layer.options.pane = panes.layerPane || 'overlayPane';
    } else if (type === 'vertexPane') {
      layer.options.pane = panes.vertexPane || 'markerPane';
    } else if (type === 'markerPane') {
      layer.options.pane = panes.markerPane || 'markerPane';
    }
  },
```